hyrule/anaphoric.hy

```diff
@@ -6,38 +6,16 @@ concise and easy to read.
     deliberately captures some form supplied to the macro which may be
     referred to by an anaphor (an expression referring to another).
 
-    -- Wikipedia (https://en.wikipedia.org/wiki/Anaphoric_macro)
-
-These macros are implemented by replacing any use of the designated
-anaphoric symbols (``it``, in most cases) with a gensym. Consequently,
-it's unwise to nest these macros where symbol replacement is happening.
-Symbol replacement typically takes place in ``body`` or ``form``
-parameters, where the output of the expression may be returned. It is also
-recommended to avoid using an affected symbol as something other than a
-variable name, as in ``(print \"My favorite Stephen King book is\" 'it)``."
+    -- Wikipedia (https://en.wikipedia.org/wiki/Anaphoric_macro)"
 
 (require
   hyrule.macrotools [defmacro!]
   hyrule.argmove [->])
 
-;;; Macro to help write anaphoric macros
-
-(defmacro rit [#* body]
-  "Supply `it` as a gensym and R as a function to replace `it` with the
-  given gensym throughout expressions."
-  `(do
-    (setv it (hy.gensym))
-    (defn R [form]
-      "Replace `it` with a gensym throughout `form`."
-      (recur-sym-replace {'it it} form))
-    ~@body))
-
-
-;;; These macros make writing functional programs more concise
 
 (defmacro ap-if [test-form then-form [else-form None]]
```

Comment:
do we want to take to make the else form required given that core `if` it's required now? We'd probably have to add an `ap-when`to fill the gap though

Reply:
I thought about it, but I figured it wasn't super-necessary. These macros take some inspiration from core macros, but they aren't quite the same. Maybe somebody who uses these guys a lot will want to add `ap-cond` and `ap-when`. But I don't use 'em.

Reply:
I'd be happy to make them. I end up using them from time to time. I'll give it a test and see how it feels and maybe open a second PR. It would probably be it of scope in this one anyways

Reply:
Oops, I merged this just as you were writing your comment. But yeah, it could be a new PR.